lib/model/MessageType.d.ts

```diff
@@ -0,0 +1,83 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ramiro González Maciel <ramiro@modusbox.com>
+
+ --------------
+ ******/
+declare enum EnEventType {
+    log = "log",
+    audit = "audit",
+    error = "error",
+    trace = "trace"
+}
+declare enum EnEventTypeAction {
```

Comment:
I would recommend instead defining an object for each of the event types which contain the respective enums for each of their actions instead of being a general list of actions which may or may not be applicable to certain events.

E.g. `debug`, `info`, etc are only applicable to the `log` event type.

The downside is that you will most likely have to define several enum actions for each type, but I believe this would remove any confusion.

Let me know your thoughts here.

Also, let me know about this.

I originally used the prefix of `En` for Enum and `Ob` for object. I wonder if we should rather change enums into CAPS and remove the prefixes from the implementation. I will update the documentation to reflect this change if you agree with it. The prefixes are looking for verbose to me.

Reply:
I agree on removing the prefixes. I'd also like to remove the "Type" suffix in `MessageType` and other messages/classes. In `MessageType` I think it's redundant because we're defining a class. And it is confusing because we have `EnEventType` were we use it to represent the kind(enum) of Event. 

I'll try to define the types/actions so as to avoid having invalid combinations.

Reply:
> I agree on removing the prefixes. I'd also like to remove the "Type" suffix in `MessageType` and other messages/classes. In `MessageType` I think it's redundant because we're defining a class. And it is confusing because we have `EnEventType` were we use it to represent the kind(enum) of Event.
> 
> I'll try to define the types/actions so as to avoid having invalid combinations.

Cool. Let's figure out the details in tomorrow's meeting.

Reply:
I've made the changes in the new commits below so we can review them later